src/ifvc.c

```diff
@@ -386,7 +260,7 @@ struct IfDesc *getIfByAddress( uint32_t ipaddr ) {
 *   the supplied IP adress. The IP must match a interfaces
 *   subnet, or any configured allowed subnet on a interface.
 */
-struct IfDesc *getIfByVifIndex( unsigned vifindex ) {
+struct IfDesc *getIfByVifIndex( signed vifindex ) {
```

Comment:
What does mean negative vifindex?

Reply:
This was not my invention. In current version, array of upstream interfaces is initialized as          // init array to "not set"

        for ( Ix = 0; Ix < MAX_UPS_VIFS; Ix++)

        {

            upStreamIfIdx[Ix] = -1;

        }

It was however never used to compare old interfaces array to new interfaces, so that's why compiler now throws warning.

Reply:
ok!

Reply:
anyway, you should use `int` type. `signed` is just alias for `int` and `int` is more common.

Reply:
switched the declarations to int